src/utils/getResponse.ts

```diff
@@ -23,14 +23,18 @@ export const getResponse = async <
   req: R,
   handlers: H,
 ): Promise<ResponsePayload> => {
-  const [relevantHandler, parsedRequest]: [
+  const [relevantHandler, parsedRequest, mockedResponse, publicRequest]: [
     RequestHandler<any, any>,
     R,
-  ] = handlers.reduce<any>((found, requestHandler) => {
+    MockedResponse | null,
+    any,
+  ] = (await handlers.reduce<any>(async (handlerPromise, requestHandler) => {
```

Comment:
The approach used there is not good of course. 
If two requests return something what we'll do?

Reply:
If there is a request handler that returns a mocked response (or `null`), then it must stop the iteration, regardless if there are any other request handlers after it. If a request handler does not have a return statement, MSW should still execute its resolver.

So, it can be summarized as: MSW executes all request handlers which predicate matches the captured request _until_ any of them returns something.